src/pages/instances/InstanceList.tsx

```diff
@@ -189,8 +193,103 @@ const InstanceList: FC = () => {
         typeof item.content !== "string" || !hiddenCols.includes(item.content)
     );
 
-  const getRows = (hiddenCols: string[]) =>
-    filteredInstances.map((instance) => {
+  const { data: operationList } = useQuery({
+    queryKey: [queryKeys.operations, project],
+    queryFn: () => fetchOperations(project),
+    refetchInterval: 1000,
+  });
+
+  if (error) {
+    notify.failure("Loading operations failed", error);
+  }
+
+  const creationOperations = (operationList?.running ?? []).filter(
+    (item) => item.description === "Creating instance"
+  );
+
+  const getRows = (hiddenCols: string[]): MainTableRow[] => {
+    let spannedWidth = 870;
+    if (hiddenCols.includes(TYPE)) {
+      spannedWidth -= 130;
+    }
+    if (hiddenCols.includes(DESCRIPTION)) {
+      spannedWidth -= 150;
+    }
+    if (hiddenCols.includes(IPV4)) {
+      spannedWidth -= 150;
+    }
+    if (hiddenCols.includes(IPV6)) {
+      spannedWidth -= 330;
+    }
+    if (hiddenCols.includes(SNAPSHOTS)) {
+      spannedWidth -= 110;
+    }
+    if (hiddenCols.includes(STATUS)) {
+      spannedWidth -= 160;
```

Comment:
I agree, this is quite ugly. This shadows the values from [instance_list.scss](https://github.com/canonical/lxd-ui/blob/main/src/sass/_instance_list.scss#L188) for the info cell to mimic the same width as the original columns.

We can't use `getBoundingClientRect`, because all columns will be stretched according to the current screen width. So it would give a different value and the rendered info column width wouldn't match the other columns it is spanning over.

Reply:
Since we cannot do things in an elegant way, let's consider this problem: there's a high chance that if for some reason in the future one changes the values in the scss, they will forget/not know that they need to update them also here.

So if we cannot really do anything about the hard-coded values, my proposal is:

1. Move all these values to some `const` that can be referenced when needed, e.g.:
`const tableWidths = { type: "130px", ... }`
3. Get rid of the CSS classes that only specify a width for these columns, e.g.:
~~`.type { width: 130px; }`~~
5. Use the `const` values here and to set an inline width to headers and rows, e.g.:
`... { content: TYPE, sortKey: "type", style: { width: tableWidths.type } }, ...` (header)
`..., 
className: "clickable-cell",
style: { width: tableWidths.type }, ...
` (row)

Reply:
Also, I'm still experiencing some weird behaviour that I think might be related to this width calculation:

![Peek 2023-05-23 09-56](https://github.com/canonical/lxd-ui/assets/56583786/ccd153ba-fdd7-4a66-b6e2-8550ccdf1157)

For reference, the `window.innerWidth` here is 1517.

Reply:
Good, what about the other width issue mentioned above?

![Screenshot from 2023-05-23 10:39:50](https://github.com/canonical/lxd-ui/assets/56583786/4d144d75-e45b-48be-abb2-4af2e18f542d)

![Screenshot from 2023-05-23 10:40:09](https://github.com/canonical/lxd-ui/assets/56583786/90d0e19d-9bda-4947-9c80-81acecaa4a28)

Reply:
I think this was due to a missing dependency to the `creationOperations` in a `useEffect`. Just pushed a fix. Please reopen this if it still happens.

Reply:
Yeah it's still happening after building with latest changes. It might not be related to table height, but rather to the width calculation. Can you try reproduce it yourself? Set a 1517 width and try to create an instance in a project with zero instances, you should see this:

![Screenshot from 2023-05-23 11:01:59](https://github.com/canonical/lxd-ui/assets/56583786/26e38366-e25e-4ba0-a7c9-6da451aa9d3f)

Reply:
Thanks for the explainer, can now reproduce it and will find a fix.

Reply:
Pushed a fix for it.